src/deploy/creation_common.rs

```diff
@@ -235,6 +236,33 @@ pub(super) fn payment_str_params(matches: &ArgMatches) -> PaymentStrParams<'_> {
     unreachable!("clap arg groups and parsing should prevent this")
 }
 
+/// Handles providing the arg for speculative execution.
+pub(super) mod speculative_exec {
+    use super::*;
+
+    const ARG_NAME: &str = "speculative_exec";
+    const ARG_VALUE_NAME: &str = "HEX STRING OR INTEGER";
+    const ARG_HELP: &str =
+        "The endpoint to perform the speculative execution, identified by a BlockIdentifier, \
+        which is either a cryptographic hash or a u64.";
+
+    pub(in crate::deploy) fn arg() -> Arg {
+        Arg::new(ARG_NAME)
+            .long(ARG_NAME)
+            .required(false)
+            .value_name(ARG_VALUE_NAME)
+            .help(ARG_HELP)
+            .display_order(DisplayOrder::BlockIdentifier as usize)
+    }
+
+    pub(in crate::deploy) fn get(matches: &ArgMatches) -> &str {
+        matches
+            .get_one::<String>(ARG_NAME)
+            .map(String::as_str)
+            .unwrap_or_default()
+    }
```

Comment:
I think we need to change the return type here to represent the three cases: `--speculative-exec` wasn't provided, `--speculative-exec` was provided with no value, and `--speculative-exec` was provided with a value:
```suggestion
    pub(in crate::deploy) fn get(matches: &ArgMatches) -> (bool, &str) {
        let value = matches
            .get_one::<String>(ARG_NAME)
            .map(String::as_str)
            .unwrap_or_default();
        match value {
            "" => (false, ""),
            DEFAULT_MISSING_VALUE => (true, ""),
            _ => (true, value),
        }
    }
```

Reply:
I don't understand why it doesn't work with

```Rust
    const DEFAULT_MISSING_VALUE: &str = "";
```

If that gets put in why isn't it `Some("")`?

Anyways, I changed as per your suggestion.